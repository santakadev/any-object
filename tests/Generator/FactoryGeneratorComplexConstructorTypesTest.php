<?php

namespace Santakadev\AnyObject\Tests\Generator;

use ApprovalTests\Approvals;
use Santakadev\AnyObject\Tests\Generator\Generated\AnyNamedConstructorObject;
use Santakadev\AnyObject\Tests\TestData\ComplexContructorTypes\NamedConstructorObject;

class FactoryGeneratorComplexConstructorTypesTest extends FactoryGeneratorTestCase
{
    public function test_generator_from_named_constructor(): void
    {
        $this->generateFactoryFor(NamedConstructorObject::class);

        $text = $this->readGeneratedAnyFileFor(NamedConstructorObject::class);
        Approvals::verifyString($text);
        $test = AnyNamedConstructorObject::with(value: "test");
        $this->assertEquals("test", $test->value);
    }
}
