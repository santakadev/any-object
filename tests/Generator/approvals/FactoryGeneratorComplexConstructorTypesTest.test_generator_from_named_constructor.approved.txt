<?php

namespace Santakadev\AnyObject\Tests\Generator\Generated;

use Faker\Factory;
use Santakadev\AnyObject\Tests\TestData\ComplexContructorTypes\NamedConstructorObject;

final class AnyNamedConstructorObject
{
    public static function with(string|ValueNotProvided $value = new ValueNotProvided()) : NamedConstructorObject
    {
        if ($value instanceof ValueNotProvided) {
            $faker = Factory::create();
            $value = $faker->text(200);
        }
        return NamedConstructorObject::fromString($value);
    }
    public static function build() : NamedConstructorObject
    {
        return self::with();
    }
}
